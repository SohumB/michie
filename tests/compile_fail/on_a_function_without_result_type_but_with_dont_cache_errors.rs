use michie::memoized;

#[memoized(key_expr = (), dont_cache_errors)]
fn f() -> i32 { 4 }

fn main() {}
