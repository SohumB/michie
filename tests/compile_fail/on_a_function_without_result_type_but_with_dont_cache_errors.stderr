error: function return type must be Result<T,E> (for some T, E)
 --> tests/compile_fail/on_a_function_without_result_type_but_with_dont_cache_errors.rs:3:1
  |
3 | #[memoized(key_expr = (), dont_cache_errors)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)
